operators/src/pro/machine_learning/onnx.rs

```diff
@@ -0,0 +1,473 @@
+            name,
+            result_descriptor: out_descriptor,
+            source,
+        }))
+    }
+
+    span_fn!(Onnx);
+}
+
+pub struct InitializedOnnx {
+    name: CanonicOperatorName,
+    result_descriptor: RasterResultDescriptor,
+    source: Box<dyn InitializedRasterOperator>,
+}
+
+impl InitializedRasterOperator for InitializedOnnx {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::I64(
+            // TODO: output type depends on ml task, e.g. classification or regression
+            OnnxProcessor::new(
+                self.source.query_processor()?.into_f32(),
+                self.result_descriptor.clone(),
+            )
+            .boxed(),
+        ))
+    }
+
+    fn canonic_name(&self) -> CanonicOperatorName {
+        self.name.clone()
+    }
+}
+
+pub(crate) struct OnnxProcessor {
+    source: Box<dyn RasterQueryProcessor<RasterType = f32>>, // as most ml algorithms work on f32 we use this as input type
+    result_descriptor: RasterResultDescriptor,
+}
+
+impl OnnxProcessor {
+    pub fn new(
+        source: Box<dyn RasterQueryProcessor<RasterType = f32>>,
+        result_descriptor: RasterResultDescriptor,
+    ) -> Self {
+        Self {
+            source,
+            result_descriptor,
+        }
+    }
+}
+
+#[async_trait]
+impl RasterQueryProcessor for OnnxProcessor {
+    type RasterType = i64; // TODO: output type depends on ml task, e.g. classification or regression
+
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<BoxStream<'a, Result<RasterTile2D<i64>>>> {
+        let num_bands = self.source.raster_result_descriptor().bands.count() as usize;
+
+        let mut source_query = query.clone();
+        source_query.attributes = (0..num_bands as u32).collect::<Vec<u32>>().try_into()?;
+
+        let session = ort::Session::builder()
+            .context(error::Onnx)?
+            .commit_from_file(test_data!("pro/ml/onnx/test.onnx"))
+            .context(error::Onnx)?;
+
+        let stream = self
+            .source
+            .raster_query(source_query, ctx)
+            .await?
+            .chunks(num_bands) // chunk the tiles to get all bands for a spatial index at once
+            .map(move |chunk| {
+                if chunk.len() != num_bands {
+                    // if there are not exactly N tiles, it should mean the last tile was an error and the chunker ended prematurely
+                    if let Some(Err(e)) = chunk.into_iter().last() {
+                        return Err(e);
+                    }
+                    // if there is no error, the source did not produce all bands, which likely means a bug in an operator
+                    unreachable!("the source did not produce all bands");
```

Comment:
i guess it should be an error not unreachable? I tought unreachable must never happen. Maybe also add a warn! log here?